PowerShell/JumpCloud Module/Public/Users/Remove-JCUser.ps1

```diff
@@ -67,52 +92,106 @@ UserID has an Alias of _id. This means you can leverage the PowerShell pipeline
+                Switch ($CascadeManager) {
+                    'NULL' {
+                        $Status = Delete-JCUser -Id $UserID -managerId $null -Headers $hdrs -UserHash $UserHash
+                    }
+                    'Auto' {
+                        if ($hasManagerId) {
+                            $Status = Delete-JCUser -Id $UserID -managerId $hasManagerId -Headers $hdrs -UserHash $UserHash
+                        } else {
+                            $Status = Delete-JCUser -Id $UserID -managerId $null -Headers $hdrs -UserHash $UserHash
+                        }
+                    }
+                    'ID' {
+                        $Status = Delete-JCUser -Id $UserID -managerId $CascadeManagerId -Headers $hdrs -UserHash $UserHash
+                    }
+                }
+            } elseif ($isManager -and !$CascadeManager) {
+                # Prompt for CascadeManager, user enters the ID of the new manager
+                $cascade_manager = Read-Host "User $($Username) is a manager. Do you want to reassign their managed users to another manager? (Y / N)"
+                if ($cascade_manager -eq 'Y') {
+                    if ($hasManagerId) {
+                        $managerUsername = $UserHash.GetEnumerator().Where({ $_.Name -contains ($hasManagerId) }) | Select-Object -ExpandProperty Value | Select-Object -ExpandProperty username
+                        $cascade_manager = Read-Host "User $($Username) is managed by manager: $($managerUsername). Do you want to reassign the managed users to the manager: $($managerUsername)? (Y/N)"
+                        if ($cascade_manager -eq 'Y') {
+                            $newManagerId = $hasManagerId
+                            $Status = Delete-JCUser -Id $UserID -managerId $newManagerId -Headers $hdrs -UserHash $UserHash
+                        } elseif ($cascade_manager -eq 'N') {
+                            $newManagerId = Read-Host "Enter the UserID of the new manager"
+                            # Validate if the Id is a JC User
+                            try {
+                                $validateUser = Get-JcSdkUser -Id $newManagerId
+                                Write-Debug "User $newManagerId is a valid JumpCloud User"
+                                $Status = Delete-JCUser -Id $UserID -managerId $newManagerId -Headers $hdrs -UserHash $UserHash
+                            } catch {
+                                Write-Error "User does not exist. Please enter a valid UserID."
+                                # Exit the script
+                                Exit
+                            }
+                        }
+                    } else {
+                        $newManagerId = Read-Host "Enter the UserID of the new manager"
+                        # Validate if the Id is a JC User
+                        try {
+                            $validateUser = Get-JcSdkUser -Id $newManagerId
```

Comment:
Every user is already in the hash on line 74, we should avoid making another unnecessary call to validate the user here